core/rawdb/accessors_indexes.go

```diff
@@ -485,3 +450,24 @@ func DeleteFilterMapsRange(db ethdb.KeyValueWriter) {
 		log.Crit("Failed to delete filter maps range", "err", err)
 	}
 }
+
+// deletePrefixRange deletes everything with the given prefix from the database.
+func deletePrefixRange(db ethdb.KeyValueRangeDeleter, prefix []byte) error {
+	end := bytes.Clone(prefix)
+	end[len(end)-1]++
```

Comment:
It's not 100% correct, but probably good enough for the normal cases.

The issue is if the last byte of prefix is 0xff, then the range will be invalid.

```go
// increaseKey increase the input key by one bit. Return nil if the entire
// addition operation overflows.
func increaseKey(key []byte) []byte {
	for i := len(key) - 1; i >= 0; i-- {
		key[i]++
		if key[i] != 0x0 {
			return key
		}
	}
	return nil
}
```

Reply:
I know but is is never 0xff, it is a constant we specify in schema.go.